### PR TITLE
Implement IDisposable in RedisCacheWrapper

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Redis/Services/RedisCacheWrapper.cs [hunk 1]
```diff
@@ -8,12 +8,16 @@ namespace OrchardCore.Redis.Services;
...
 /// Wrapper preventing the <see cref="RedisCache"/> to dispose a shared <see cref="IConnectionMultiplexer"/>.
 /// </summary>
-public class RedisCacheWrapper : IDistributedCache
+public class RedisCacheWrapper : IDistributedCache, IDisposable
 {
     private readonly RedisCache _cache;
 
     public RedisCacheWrapper(IOptions<RedisCacheOptions> optionsAccessor) => _cache = new RedisCache(optionsAccessor);
 
+#pragma warning disable CA1816
```
Comment (gvkries) [inline src/OrchardCore.Modules/OrchardCore.Redis/Services/RedisCacheWrapper.cs:17]: This warning should not be disabled, either call `GC.SuppressFinalize` or seal the class (which I think is preferred here).

**Review by gvkries — CHANGES_REQUESTED** [review]

File: src/OrchardCore.Modules/OrchardCore.Redis/Services/RedisCacheWrapper.cs (hunk 1 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Redis/Services/RedisCacheWrapper.cs:17]: I was thinking of implementing a disposing pattern, but seal the class is fine

**Review by gvkries — APPROVED** [review]

**Comment by sebastienros** [comment]
I think it's important to explain the intent of such changes, at least sharing what the scopes of the service is and why this change is better. So I dug a little... 

The wrapper is a singleton, at first I didn't see why calling Dispose is better in any way. Being a singleton everything will be release once the application stops.

However I looked into the history of this "Wrapper". And it seems that it was added to prevent the internal redis cache service to be "DIsposed" when a tenant is restarted, or the IConnectionMultiplexer that is shared across tenants would be disposed too. Hence the goal of this wrapper is to NOT dispose the redis cache.

Please revert it an add a comment on this class explain what its purpose is and why the redis cache should not be disposed. IF there were other goals behind this change please explain them.

**Comment by hishamco** [comment]
@sebastienros  yes the intention is not to dispose of the cache because it's shared across tenants, I don't think there's wrong with disposing of the cache once the Wrapper is disposed

For instance, we disposed of the list of shells once the `ShellHost` is disposed which is make sense to me

**Comment by sebastienros** [comment]
> I don't think there's wrong with disposing of the cache once the Wrapper is disposed

I literally explained why it's wrong in the comment I made

**Comment by hishamco** [comment]
I will revert the commit, but at least we should right this class as `sealed`

**Comment by sebastienros** [comment]
> I will revert the commit, but at least we should mark this class as sealed, right?

Yes this is fine, in your someone will also add the comments on the class to explain why it should not dispose the redis cache.

**Pull request merged**